source/preprocessor.tex

```diff
@@ -222,9 +222,8 @@
 \pnum
 At the start of phase 4 of translation,
 the \grammarterm{group} of a \grammarterm{pp-global-module-fragment} shall
-neither contain a \grammarterm{control-line}
-not starting with a \tcode{\#} preprocessing token,
-nor contain a \grammarterm{text-line}.
+contain neither a \grammarterm{text-line} nor a \grammarterm{control-line}
+not starting with a \tcode{\#} preprocessing token.
```

Comment:
The new wording has ambiguous grouping: to what does the "not starting with" attach? (Now, I know that it doesn't matter, because a *text-line* can't start with a `#`, but we should still try to avoid ambiguities even if they don't matter.)

Reply:
Wouldn't it have to be "neither X, nor Y, not starting with Z" to be ambiguous?

Reply:
I don't think the lack of commas disambiguates it sufficiently. I find the revised sentence substantially harder to read than the original. How about this:

> At the start of phase 4 of translation, the *group* of a *pp-global-module-fragment* shall not contain a *text-line* nor a *pp-import*.

Reply:
Good; I've done that.  If we ever add more non-`#` directives, we can introduce a collective grammar term for them.